python/pyspark/sql/types.py

```diff
@@ -176,14 +177,36 @@ def needConversion(self):
 
     def toInternal(self, dt):
         if dt is not None:
-            seconds = (calendar.timegm(dt.utctimetuple()) if dt.tzinfo
-                       else time.mktime(dt.timetuple()))
+            if platform.system().lower() == "windows":
+                """
+                On Windows, the current value is converted to a timestamp
+                when the current value is less than 1970
+                """
```

Comment:
docstring should be placed at the top of the function.

I think we can just leave the multi-line comments in this case.